unittests/CMakeLists.txt

```diff
@@ -1,4 +1,31 @@
 
+if(SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER)
+  if((NOT "${CMAKE_C_COMPILER_ID}" STREQUAL "Clang") AND
+     (NOT "${CMAKE_C_COMPILER_ID}" STREQUAL "AppleClang"))
+    message(FATAL_ERROR "Building the swift runtime is not supported with ${CMAKE_C_COMPILER_ID}. Use the just-built clang instead.")
+  else()
+    message(WARNING "Building the swift runtime using the host compiler, and not the just-built clang.")
+  endif()
+else()
+  # If we use Clang-cl or MSVC, CMake provides default compiler and linker flags that are incompatible
+  # with the frontend of Clang or Clang++.
+  if(SWIFT_COMPILER_IS_MSVC_LIKE)
+    set(CMAKE_CXX_COMPILER "${SWIFT_NATIVE_LLVM_TOOLS_PATH}/clang-cl")
+    set(CMAKE_C_COMPILER "${SWIFT_NATIVE_LLVM_TOOLS_PATH}/clang-cl")
+  else()
+    set(CMAKE_CXX_COMPILER "${SWIFT_NATIVE_LLVM_TOOLS_PATH}/clang++")
+    set(CMAKE_C_COMPILER "${SWIFT_NATIVE_LLVM_TOOLS_PATH}/clang")
+  endif()
+  
+  set(CMAKE_CXX_COMPILER_ARG1 "")
+  set(CMAKE_C_COMPILER_ARG1 "")
+  # The sanitizers require using the same version of the compiler for
+  # everything and there are various places where we link runtime code with
+  # code built by the host compiler. Disable sanitizers for the runtime for
+  # now.
+  append("-fno-sanitize=all" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
```

Comment:
I'm a bit confused! Like above, this is what `stdlib/CMakeLists.txt` does. Then `stdlib/public/CMakeLists.txt` checks for `SWIFT_RUNTIME_USE_SANITIZERS` and sets `fsanitize=thread`. Does the deeper one override it?

Reply:
Yes, the latter one wins.

Reply:
Thanks for clarifying. I'll pull in that logic from `stdlib/public/CMakeLists.txt`. There must be a way to share this stuff instead of duplicating it, but I fear it's beyond me.